libkineto/src/output_json.cpp

```diff
@@ -39,22 +39,23 @@ void ChromeTraceLogger::handleTraceStart(
   )JSON", kSchemaVersion);
 
   if (!metadata.empty()) {
-    traceOf_ << R"JSON(
-  "metadata": {
-  )JSON";
-    bool first = true;
+    std::vector<std::string> vals;
     for (const auto& kv : metadata) {
-      if (!first) {
-        traceOf_ << ",\n";
-      }
-      traceOf_ << fmt::format(R"(    "{}": "{}")", kv.first, kv.second);
-      first = false;
+      vals.push_back(fmt::format(R"(    "{}": "{}")", kv.first, kv.second));
     }
-    traceOf_ << R"JSON(
-  },
-  )JSON";
+    traceOf_ << fmt::format(R"JSON(
+  "metadata": {{
+    {}
+  }},)JSON", fmt::join(vals, ",\n"));
   }
 
+#ifdef HAS_CUPTI
+  traceOf_ << fmt::format(R"JSON(
+  "computeProperties": [
```

Comment:
need confirm with @gdankel about using top level property or putting it into metadata.

My personal preference is put it into metadata.